crates/rethnet_evm/benches/state.rs

```diff
@@ -115,5 +127,21 @@ fn bench_basic(c: &mut Criterion) {
     });
 }
 
-criterion_group!(benches, bench_insert_account, bench_checkpoint, bench_basic);
+fn bench_code_by_hash(c: &mut Criterion) {
+    bench_sync_state_method(c, "StateRef::code_by_hash", |state, number_of_accounts| {
+        for i in (1..=number_of_accounts).rev() {
+            state
+                .code_by_hash(Bytecode::new_raw(Bytes::from(format!("0x{i:x}"))).hash())
```

Comment:
Using the improvement mentioned above, combined with `Address::from_low_u64_ne`, you can make this more performant, reducing the overhead that string operations will have on your benchmark

Reply:
Similar to the other comment, I'm confused about what you're looking for here. It's important that the object on which `.hash()` is called here is constructed the same exact way as done in `fill()` (where you left your other comment), so that the input will be a hash whose code is actually present.

Reply:
Addressed in https://github.com/NomicFoundation/hardhat/pull/3881/commits/2074b056344a742c870a3afd12c4efd5b9c8a559.

Reply:
This is what I had in mind:

```rust
Bytecode::new_raw(Bytes::copy_from_slice(address.as_bytes()))
```